### PR TITLE
all_apparmor_profiles_in_enforce_complain_mode: Fix OVAL logic

### PR DESCRIPTION
**Pull request by dodys**
#### Description:

- Current OVAL fails with `unknown` result because the variables are looking for a `subexpression` field of the object when there's none. Instead use `text` part of the object. Also change how you seek for unconfined.
- Also fixes test not working on sle15.

#### Rationale:

- To have a `subexpression` field from the object you would need to select some specific text with parenthesis, but that's not being done. Since we don't care about the actual pattern being matched but instead the number of matches, using the `text` field instead is enough.

#### Review Hints:

- OVAL passes.

### CODE REVIEW

**Comment by dodys** [comment]
@teacup-on-rockingchair is the tests you added to this rule actually working? they seem completely broken right now. I tried to fix it by adding the apparmor-utils to platform-packages-override and now it fails with something else.
I will leave it to you to add what is missing there and to test my changes.

**Comment by teacup-on-rockingchair** [comment]
> @teacup-on-rockingchair is the tests you added to this rule actually working? they seem completely broken right now. I tried to fix it by adding the apparmor-utils to platform-packages-override and now it fails with something else. I will leave it to you to add what is missing there and to test my changes.

I am afraid for the aa-teardown, and apparmor_parser used in the tests you need to add also `apparmor-parser` package, but anyways the tools in the tests require appropriate kernel patches etc, so should not consider them ready for container environment.

**Comment by dodys** [comment]
> > @teacup-on-rockingchair is the tests you added to this rule actually working? they seem completely broken right now. I tried to fix it by adding the apparmor-utils to platform-packages-override and now it fails with something else. I will leave it to you to add what is missing there and to test my changes.
> 
> I am afraid for the aa-teardown, and apparmor_parser used in the tests you need to add also `apparmor-parser` package, but anyways the tools in the tests require appropriate kernel patches etc, so should not consider them ready for container environment.

in that case, we should waiver the test failures.

Could you please check now if it works fine on sles?

File: linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/oval/shared.xml [hunk 1]
```diff
@@ -24,32 +24,32 @@
...
+      <object_component item_field="text"
                         object_ref="{{{ rule_id }}}_obj_apparmor_profiles" />
     </count>
   </local_variable>
   <local_variable datatype="int" comment="enforced apparmor profiles"
     id="{{{ rule_id }}}_var_num_apparmor_enforced_complaining_profiles" version="1" >
     <arithmetic arithmetic_operation="add">
       <count>
-      <object_component item_field="subexpression"
+      <object_component item_field="text"
```
[the remark this answers is not in the document]
Comment (teacup-on-rockingchair) [reply linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/oval/shared.xml]: I think you are right we will be on much simpler track if we just check for any unconfined :bow: 
Not sure what was going in my mind playing with arithmetic looked so tempting ;)

**Comment by teacup-on-rockingchair** [comment]
> > > @teacup-on-rockingchair is the tests you added to this rule actually working? they seem completely broken right now. I tried to fix it by adding the apparmor-utils to platform-packages-override and now it fails with something else. I will leave it to you to add what is missing there and to test my changes.
> > 
> > 
> > I am afraid for the aa-teardown, and apparmor_parser used in the tests you need to add also `apparmor-parser` package, but anyways the tools in the tests require appropriate kernel patches etc, so should not consider them ready for container environment.
> 
> in that case, we should waiver the test failures.
> 
> Could you please check now if it works fine on sles?

Yes they work with the patches and are failing on master branch with my half baked commit so I guess we are ok there :)

**Review by teacup-on-rockingchair — APPROVED** [review]
Thanks for the contribution and corrections @dodys 🙇  . If you decide to go with the suggested more simplistic approach just ping me to approve it again. Overall I think most important is that we have a working precedent of oval check working with apparmor ;)

**Comment by dodys** [comment]
> Thanks for the contribution and corrections @dodys 🙇 . If you decide to go with the suggested more simplistic approach just ping me to approve it again. Overall I think most important is that we have a working precedent of oval check working with apparmor ;)

I feel like this OVAL still doesn't implement the whole check from CIS. It only checks profiles, not processes. 
I was mostly fixing it so Ubuntu can still rely on the SCE check instead.
I will leave it as-is for now. If you could merge it, it would be great :)
Then any later work on this OVAL can be done in a next PR.

**Pull request merged**